flow360/component/volume_mesh.py

```diff
@@ -772,3 +786,192 @@ def __getitem__(self, index) -> VolumeMesh:
+            )
+
+        if self.solver_version is None:
+            raise Flow360ValueError("solver_version field is required.")
+
+    # pylint: disable=protected-access
+    def submit(self, description="", progress_callback=None) -> VolumeMeshV2:
+        """
+        Submit volume mesh to cloud and create a new project
+
+        Parameters
+        ----------
+        description : str, optional
+            description of the project, by default ""
+        progress_callback : callback, optional
+            Use for custom progress bar, by default None
+
+        Returns
+        -------
+        VolumeMeshV2
+            Volume mesh object with id
+        """
+
+        self._validate()
+
+        if not shared_account_confirm_proceed():
+            raise Flow360ValueError("User aborted resource submit.")
+
+        req = NewVolumeMeshRequestV2(
+            name=self.project_name,
+            solver_version=self.solver_version,
+            tags=self.tags,
+            file_name=os.path.basename(self.file_name),
+            # pylint: disable=fixme
+            # TODO: remove hardcoding
+            parent_folder_id="ROOT.FLOW360",
+            length_unit=self.length_unit,
+            format=self.format.value,
+            description=description,
+        )
+
+        ##:: Create new Volume mesh resource and project
+        resp = RestApi(VolumeMeshInterfaceV2.endpoint).post(req.dict())
+        info = VolumeMeshMetaV2(**resp)
+
+        ##:: upload volume mesh file
+        volume_mesh = VolumeMeshV2(info.id)
+        heartbeat_info = {"resourceId": info.id, "resourceType": "VolumeMesh", "stop": False}
+        # Keep posting the heartbeat to keep server patient about uploading.
+        heartbeat_thread = threading.Thread(target=post_upload_heartbeat, args=(heartbeat_info,))
+        heartbeat_thread.start()
+        volume_mesh._webapi._upload_file(
+            remote_file_name=os.path.basename(self.file_name),
+            file_name=self.file_name,
+            progress_callback=progress_callback,
+        )
+        heartbeat_info["stop"] = True
+        heartbeat_thread.join()
+        ##:: kick off pipeline
+        volume_mesh._webapi._complete_upload()
+        log.debug("Waiting for volume mesh to be processed.")
+        volume_mesh._webapi.get_info()
+        log.info("Volume mesh successfully submitted.")
+        return volume_mesh
+
+
+class VolumeMeshV2(AssetBase):
+    """
+    Volume mesh component for workbench (simulation V2)
+    """
+
+    _interface_class = VolumeMeshInterfaceV2
+    _meta_class = VolumeMeshMetaV2
+    _draft_class = VolumeMeshDraftV2
+    _web_api_class = Flow360Resource
+    _entity_info_class = VolumeMeshEntityInfo
+
+    @classmethod
+    # pylint: disable=redefined-builtin
+    def from_cloud(cls, id: str):
+        """Create asset with the given ID"""
+        asset_obj = super().from_cloud(id)
+
+        # pylint: disable=fixme
+        # TODO: Grouping logic?
```

Comment:
no grouping required for VM. But we need `[]` access to be able to select boundaries:
https://github.com/flexcompute/Flow360/blob/f6599cf7d1f50240f48d24e7ffd90c6da9cf8ab0/flow360/component/geometry.py#L397

also, we need method for getting all boundaries.

Reply:
Added in [6cfcff4](https://github.com/flexcompute/Flow360/pull/470/commits/6cfcff4a6a67046ac7f077608e592c0d088effc6)